compose/compose.go

```diff
@@ -0,0 +1,100 @@
+package compose
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/compose-spec/compose-go/cli"
+	"github.com/compose-spec/compose-go/types"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/flags"
+	"github.com/docker/compose-cli/pkg/api"
+	"github.com/docker/compose-cli/pkg/compose"
+	libstack "github.com/portainer/docker-compose-wrapper"
+)
+
+type ComposeDeployer struct {
+	configPath string
+}
+
+func NewComposeDeployer(configPath string) (libstack.Deployer, error) {
+	return &ComposeDeployer{configPath}, nil
+}
+
+// Up creates and starts containers
+func (deployer *ComposeDeployer) Deploy(projectName, host string, filePaths []string, envFilePath string) error {
+	service, err := prepareService(host, deployer.configPath)
+	if err != nil {
+		return fmt.Errorf("failed creating compose service: %w", err)
+	}
+
+	project, err := prepareProject(filePaths, projectName, envFilePath)
+	if err != nil {
+		return fmt.Errorf("failed preparing project: %w", err)
+	}
+
+	err = service.Up(context.Background(), project, api.UpOptions{})
+	if err != nil {
+		return fmt.Errorf("failed deploying: %w", err)
+	}
+
+	return nil
+}
+
+// Down stops and removes containers
+func (deployer *ComposeDeployer) Remove(projectName, host string, filePaths []string) error {
```

Comment:
Same comment as above, should support passing in context to propagate it to docker API.

```go
func (deployer *ComposeDeployer) Remove(ctx context.Context, projectName, host string, filePaths []string) error {
  ...
  err = service.Down(ctx, projectName, api.DownOptions{RemoveOrphans: true})
  ...
}
```